src/components/DonationForm/DonationForm.jsx

```diff
@@ -21,106 +60,399 @@ const DonationForm = () => {
     volunteer_hours: null,
   });
 
+  useEffect(() => {
+    const fetchBusinessId = async () => {
+      if (paramId) {
+        setFormData(prevState => ({ ...prevState, business_id: paramId }));
+      } else {
+        try {
+          const businessIdResponse = await backend.get(`/businessUser/${currentUser.uid}`);
+          const fetchedBusinessId = businessIdResponse.data[0].id;
+          setBusinessId(fetchedBusinessId);
+          setFormData(prevState => ({ ...prevState, business_id: fetchedBusinessId }));
+        } catch (error) {
+          console.error('Error fetching business ID:', error);
+        }
+      }
+    };
+
+    fetchBusinessId();
+  }, []);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has missing dependencies: 'backend', 'currentUser.uid', and 'paramId'. Either include them or remove the dependency array.